### PR TITLE
Avoid setting heights twice within 0.1s

### PR DESCRIPTION
**Pull request by hanayashiki**
Fixes https://github.com/FaridSafi/react-native-gifted-chat/issues/1809

### CODE REVIEW

**Comment by sandy2008** [comment]
👍

File: src/GiftedChat.tsx [hunk 1]
```diff
@@ -594,6 +596,11 @@ class GiftedChat<TMessage extends IMessage = IMessage> extends React.Component<
 
   onKeyboardWillShow = (e: any) => {
     if (this.props.isKeyboardInternallyHandled) {
+      if (new Date().getTime() - this._lastUpdate < 100) {
```
Comment (xcarpentier) [inline src/GiftedChat.tsx:599]: Hi,
Not tested it yet but seems a good idea.
A small suggestion maybe using Date.now():
```suggestion
      if (Date.now() - this._lastUpdate < 100) {
```

File: src/GiftedChat.tsx [hunk 2]
```diff
@@ -594,6 +596,11 @@ class GiftedChat<TMessage extends IMessage = IMessage> extends React.Component<
...
+          return;
+      }
+      this._lastUpdate = new Date().getTime();
```
Comment (xcarpentier) [inline src/GiftedChat.tsx:602]: Here too :)
```suggestion
      this._lastUpdate = Date.now()
```

**Comment by hanayashiki** [comment]
Thank you for your advice!

**Comment by Hirbod** [comment]
Well this PR introduces this for me @xcarpentier @hanayashiki 
https://streamable.com/mw4eup

I think it was there for a reason. When I remove this patch using patch-package, everything works fine again

**Comment by xcarpentier** [comment]
Ok I will revert it in the next patch version.

**Comment by xcarpentier** [comment]
@Hirbod it reverts then fixed into [v016.3](https://github.com/FaridSafi/react-native-gifted-chat/releases/tag/v0.16.3)

**Comment by fierysolid** [comment]
Is this library getting new support?! Very happy right now

**Pull request merged**